### ISSUE TITLE
Feature Request: Injection values in outputName

### ISSUE THREAD

**Issue by thegoatherder** [jest-community/jest-junit#257]
```json
"jest-junit": {
  "outputName": "{suitename}.{classname}.junit.xml"
}
```

I have a complex project with 28,000 tests. At least 21,000 of the tests are data-driven - we have a process wrapped in a single `mytest.test.js` that reads data from various CSV files and tests our application using the data.

Our CI platform is Azure DevOps Pipelines. Annoyingly it does not support showing `suiteName`, `className`. Only 2 items are displayed (in a parent/child display):

- The junit filename e.g. `mytest.junit.xml`
- The `testTitle`

It's extremely difficult to see which areas of the application are failing due to the above.
I have experimented with injection values in the `titleTemplate` configuration and they do make it somewhat easier, but the team is craving a better output.

We could solve our problem in DevOps if the junit `outputName` supported injection values. It currently does not. If we use a configuration like so:

```json
"jest-junit": {
  "outputName": "{suitename}.{classname}.junit.xml"
}
```
then we get a file output with filename: `{suitename}.{classname}.junit.xml`

Is it possible to implement injection values for the `outputName`? I assume some string cleansing will be required to ensure that any characters used in the `{suitename}`, `{classname}`, etc. are allowable characters in filesystems. For example, `>`, `/`, `\` and similar chars would need to be either replaced or removed prior to injection (or Jest could just throw an error forcing developers to rename these items when used in an `outputName`)

**Comment by palmerj3** [comment]
Hey thanks for the feature request!

I agree this would be super useful.

There are some challenges we would have to overcome if we take this on.
Primarily, the challenge is that jest-junit writes a junit.xml file [when the reporter is finished](https://github.com/jest-community/jest-junit/blob/master/index.js#L32)  processing the entirety of a test report.

So, injection values like suitename or classname aren't available in that context. A report will often be a collection of suitenames and classnames, so which do we choose? How do we allow one to choose?

The thing I would ask you to consider is making heavy use of [Jest Projects](https://jestjs.io/docs/configuration#projects-arraystring--projectconfig) .
If you organize your tests utilizing projects then jest will execute each project in parallel and jest-junit will also generate a separate junit file for each project, since that is inherently how reporters work when jest is configured to use projects.

Let me know if that is a viable path forward for you, or if I should consider another way to approach this.

**Comment by thegoatherder** [comment]
Hi apologies for the late response. I took a look at Jest Projects but it didn’t suit our requirements. We have since found that Azure DevOps Test Reports does actually give a tree view of classname > test title, which gives us what we need for now. It would be even better if we could output the junit files with suite/class based filenames, but for now we have a good workaround using the 2 views on DevOps and the existing templates available!
